test/test.downscopedclient.ts

```diff
@@ -0,0 +1,406 @@
+          cabWithOnlyAvailabilityConditionExpression
+        );
+      });
+    });
+
+    it('should set custom RefreshOptions', () => {
+      const refreshOptions = {
+        eagerRefreshThresholdMillis: 5000,
+        forceRefreshOnFailure: true,
+      };
+      const cabWithOneAccessBoundaryRules = {
+        accessBoundary: {
+          accessBoundaryRules: [
+            {
+              availableResource: testAvailableResource,
+              availablePermissions: [testAvailablePermission1],
+              availabilityCondition: {
+                expression: testAvailabilityConditionExpression,
+              },
+            },
+          ],
+        },
+      };
+      const downscopedClient = new DownscopedClient(
+        client,
+        cabWithOneAccessBoundaryRules,
+        refreshOptions
+      );
+      assert.strictEqual(
+        downscopedClient.forceRefreshOnFailure,
+        refreshOptions.forceRefreshOnFailure
+      );
+      assert.strictEqual(
+        downscopedClient.eagerRefreshThresholdMillis,
+        refreshOptions.eagerRefreshThresholdMillis
+      );
+    });
+  });
+
+  describe('getAccessToken()', () => {
+    let sandbox: sinon.SinonSandbox;
+    before(() => {
+      const expectedAccessTokenResponse = {
+        token: 'subject_token',
+      };
+      sandbox = sinon.createSandbox();
+      sandbox
+        .stub(client, 'getAccessToken')
+        .resolves(expectedAccessTokenResponse);
+    });
+
+    after(() => {
+      sandbox.restore();
+    });
+
+    it('should return current unexpired cached DownscopedClient access token', async () => {
+      const now = new Date().getTime();
+      const credentials = {
+        access_token: 'DOWNSCOPED_CLIENT_ACCESS_TOKEN',
+        expiry_date: now + ONE_HOUR_IN_SECS * 1000,
+      };
+      const downscopedClient = new DownscopedClient(
+        client,
+        testClientAccessBoundary
+      );
+      downscopedClient.setCredentials(credentials);
+      const tokenResponse = await downscopedClient.getAccessToken();
+      assert.deepStrictEqual(tokenResponse.token, credentials.access_token);
+    });
+
+    it('should refresh a new DownscopedClient access when cached one gets expired', async () => {
+      const now = new Date().getTime();
+      clock = sinon.useFakeTimers(now);
+      const credentials = {
+        access_token: 'DOWNSCOPED_CLIENT_ACCESS_TOKEN',
+        expiry_date: now + ONE_HOUR_IN_SECS * 1000,
+      };
+      const scope = mockStsBetaTokenExchange([
+        {
+          statusCode: 200,
+          response: stsSuccessfulResponse,
+          request: {
+            grant_type: 'urn:ietf:params:oauth:grant-type:token-exchange',
+            requested_token_type:
+              'urn:ietf:params:oauth:token-type:access_token',
+            subject_token: 'subject_token',
+            subject_token_type: 'urn:ietf:params:oauth:token-type:access_token',
+            options:
+              testClientAccessBoundary &&
```

Comment:
You don't need this. This is always set above. It is a const. You can just directly stringify.